python/pyarrow/fs.py

```diff
@@ -183,11 +184,11 @@ def _resolve_filesystem_and_path(
     if not exists_locally:
         try:
             filesystem, path = FileSystem.from_uri(path)
-        except ValueError as e:
+        except pa.ArrowInvalid as e:
             # neither an URI nor a locally existing path, so assume that
             # local path was given and propagate a nicer file not found error
             # instead of a more confusing scheme parsing error
-            if "empty scheme" not in str(e):
+            if "URI" not in str(e):
```

Comment:
This might be too broad. I think the "Unrecognized filesystem type in URI: .." error is certainly one we want to keep raising. 
Otherwise we will assume a local file path, which will then error because of getting something that looks like a URI (and then the user might think URIs are not supported in general, while they maybe just made a typo in the URI type, or are using an unsupported file system)